src/context/WalletProvider/Coinbase/config.ts

```diff
@@ -1,9 +1,12 @@
-import { CoinbaseAdapter } from '@shapeshiftoss/hdwallet-coinbase'
 import { CoinbaseIcon } from 'components/Icons/CoinbaseIcon'
 import type { SupportedWalletInfo } from 'context/WalletProvider/config'
 
 export const CoinbaseConfig: Omit<SupportedWalletInfo, 'routes'> = {
-  adapters: [CoinbaseAdapter],
+  adapters: [
+    {
+      loadAdapter: () => import('@shapeshiftoss/hdwallet-coinbase').then(m => m.CoinbaseAdapter),
```

Comment:
we should not return `Promise<any>` here but instead create an interface for each adapter, then you can do `import type { ICoinbaseAdapter } from '@shapeshiftoss/hdwallet-coinbase'`, which would allow you to do:
```typescript
adapters: [
    {
      loadAdapter: (): Promise<ICoinbaseAdapter> =>
        import('@shapeshiftoss/hdwallet-coinbase').then(m => m.CoinbaseAdapter),
    },
  ],
```

Reply:
This is relevant to comments in the PR on top of this one, which should effectively resolve them since any types would be eliminated

Reply:
Discussed this one out, will leave it for a part 4 PR in the stack since this will require an hdwallet bumpy